### ISSUE TITLE
Allow constant expression with operator~(nint value)

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#43344]
```C#
class Program
{
    const nint x = 42;
    const nint y = ~x;
}
```
Expected: no error
Actual: `error CS0133: The expression being assigned to 'y' must be constant`
